syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -58,6 +58,42 @@ class TorchTensor(AbstractTensor):
     def has_child(self):
         return hasattr(self, "child")
 
+    def trigger_origin_backward_hook(self, sender: str, origin_id: int):
+        """
+        This hook is triggered when a tensor which was received from a sender has
+        a gradient update. It will send back to this sender and his original tensor
+        this gradient value to be set remotely. Also, because this is triggered during
+        backward(), the backward command is also forwarded back.
+
+        Args:
+            sender (str): id of the worker where this tensor comes from
+            origin_id (int): what was its original id
+        """
+
+        def trigger_origin_backward(grad):
+            """
+            The function setting back the gradient and calling backward
+
+            Args:
+                grad: the gradient tensor being set
+            """
+
+            location = self.owner.get_worker(sender)
+
+            origin_ptr = syft.PointerTensor.create_pointer(
```

Comment:
Not sure I followed...
You're creating a pointer from sender to receiver? And then set grad on this new pointer?

Reply:
I create a pointer from the receiver to the sender (so in the reverse order), back to the origin the tensor came from, and I ask to set back there a gradient

Reply:
Oh no, ok, the pointer points back to the origin tensor (thanks to id_at_location=origin_id).
So basically, you just want to send a message to sender to tell him to set_grad(something) on the tensor with id origin_id. Don't you think it would be better to use messages and actions directly instead of going through a pointer (which is less intuitive, I think)

Reply:
I'm made a proposition using your brand new TensorCommandMessage.computation :)

Reply:
Much sexier 😍